SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularitySlaveAndRackManager.java

```diff
@@ -405,7 +406,14 @@ public void checkOffer(Offer offer) {
     final SingularitySlave slave = new SingularitySlave(slaveId, host, rackId, textAttributes, Optional.<MesosResourcesObject>absent());
 
     if (check(slave, slaveManager) == CheckResult.NEW) {
-      LOG.info("Offer revealed a new slave {}", slave);
+      if (inactiveSlaves.contains(slave.getHost())) {
```

Comment:
alternatively, if we switch to the method of storing the hostnames I mentioned in the previous comment, we don't need to feed the list to this method and can just check if the individual inactive hostname node exists here